input/otlp/traces.go

```diff
@@ -468,11 +468,12 @@ func TranslateTransaction(
 		}
 
 		// Set outcome nad result from status code.
+		// However, if outcome was explicitly set to failure, preserve it
 		if statusCode := httpResponse.StatusCode; statusCode > 0 {
-			if event.Event.Outcome == outcomeUnknown {
+			if event.Event.Outcome != outcomeFailure {
 				event.Event.Outcome = serverHTTPStatusCodeOutcome(int(statusCode))
 			}
-			if event.Transaction.Result == "" {
+			if event.Transaction.Result != "Error" {
```

Comment:
Same concern as the previous one, should we be overriding an explicitly set value?